### ISSUE TITLE
When a `cf push` upload fails, a panic occurrs

### ISSUE THREAD

**Issue by lisamcho** [cloudfoundry/cli#1340]
The targeted stack appears to be up, but the upload is failing for some reason.
```
cf push
Pushing from manifest to org org / space space as admin...
Using manifest file /Users/pivotal/workspace/cf-acceptance-tests/assets/dora/manifest.yml
Getting app info...
Updating app with these attributes...
  name:                dora
  path:                /Users/pivotal/workspace/cf-acceptance-tests/assets/dora
  disk quota:          1.5G
  health check type:   port
  instances:           1
  memory:              256M
  stack:               cflinuxfs2
  routes:
    dora.beryl-fin.capi.land

Updating app dora...
Mapping routes...
Comparing local files to remote cache...
Packaging files to upload...
Uploading files...
 1.19 MiB / 1.19 MiB [=============================================================================================================================] 100.00%Retrying upload due to an error...
Uploading files...
 1.19 MiB / 1.19 MiB [=============================================================================================================================] 100.00%Retrying upload due to an error...
Uploading files...
 1.19 MiB / 1.19 MiB [=============================================================================================================================] 100.00%Retrying upload due to an error...

		Something unexpected happened. This is a bug in cf.

		Please re-run the command that caused this exception with the environment
		variable CF_TRACE set to true.

		Also, please update to the latest cli and try the command again:
		https://code.cloudfoundry.org/cli/releases

		Please create an issue at: https://code.cloudfoundry.org/cli/issues

		Include the below information when creating the issue:

		Command
		cf push

		CLI Version
		6.34.1+bbdf81482.2018-01-17

		Error
		runtime error: invalid memory address or nil pointer dereference

		Stack Trace
			goroutine 1 [running]:
	code.cloudfoundry.org/cli/util/panichandler.HandlePanic()
		/Users/worker/concourse/volumes/live/9f25a4ac-9961-42bd-7328-16fe28719986/volume/gopath/src/code.cloudfoundry.org/cli/util/panichandler/handler.go:19 +0x8a
	panic(0x192a220, 0x201dfd0)
		/usr/local/Cellar/go/1.9.2/libexec/src/runtime/panic.go:491 +0x283
	code.cloudfoundry.org/cli/command/translatableerror.UploadFailedError.Translate(0x0, 0x0, 0xc4205bc000, 0x198da01, 0x32c0080)
		/Users/worker/concourse/volumes/live/9f25a4ac-9961-42bd-7328-16fe28719986/volume/gopath/src/code.cloudfoundry.org/cli/command/translatableerror/upload_failed_error.go:16 +0x214
	code.cloudfoundry.org/cli/command/translatableerror.(*UploadFailedError).Translate(0xc420210cc0, 0xc4205bc000, 0xc420210cc0, 0x32c0080)
		<autogenerated>:1 +0x4f
	code.cloudfoundry.org/cli/util/ui.(*UI).DisplayError(0xc4205c2000, 0x1f57f60, 0xc420210cc0)
		/Users/worker/concourse/volumes/live/9f25a4ac-9961-42bd-7328-16fe28719986/volume/gopath/src/code.cloudfoundry.org/cli/util/ui/ui.go:174 +0x8b
	main.handleError(0x1f55620, 0xc4205bde60, 0x1f5dd60, 0xc4205c2000, 0x1f55620, 0xc4205bde60)
		/Users/worker/concourse/volumes/live/9f25a4ac-9961-42bd-7328-16fe28719986/volume/gopath/src/code.cloudfoundry.org/cli/main.go:257 +0x3dc
	main.executionWrapper(0x250e3c0, 0x20455a8, 0xc420198160, 0x0, 0x1, 0x0, 0x0)
		/Users/worker/concourse/volumes/live/9f25a4ac-9961-42bd-7328-16fe28719986/volume/gopath/src/code.cloudfoundry.org/cli/main.go:216 +0x26b
	code.cloudfoundry.org/cli/vendor/github.com/jessevdk/go-flags.(*Parser).ParseArgs(0xc4205b61e0, 0xc42000c090, 0x1, 0x1, 0x250bb20, 0x0, 0xc420000180, 0xc420182f60, 0x2246d90)
		/Users/worker/concourse/volumes/live/9f25a4ac-9961-42bd-7328-16fe28719986/volume/gopath/src/code.cloudfoundry.org/cli/vendor/github.com/jessevdk/go-flags/parser.go:314 +0x7fa
	main.parse(0xc42000c090, 0x1, 0x1, 0xc4205bc830)
		/Users/worker/concourse/volumes/live/9f25a4ac-9961-42bd-7328-16fe28719986/volume/gopath/src/code.cloudfoundry.org/cli/main.go:52 +0xad
	main.main()
		/Users/worker/concourse/volumes/live/9f25a4ac-9961-42bd-7328-16fe28719986/volume/gopath/src/code.cloudfoundry.org/cli/main.go:43 +0x82

	goroutine 5 [syscall]:
	os/signal.signal_recv(0x0)
		/usr/local/Cellar/go/1.9.2/libexec/src/runtime/sigqueue.go:131 +0xa7
	os/signal.loop()
		/usr/local/Cellar/go/1.9.2/libexec/src/os/signal/signal_unix.go:22 +0x22
	created by os/signal.init.0
		/usr/local/Cellar/go/1.9.2/libexec/src/os/signal/signal_unix.go:28 +0x41

	goroutine 642 [IO wait]:
	internal/poll.runtime_pollWait(0x2500f00, 0x72, 0x0)
		/usr/local/Cellar/go/1.9.2/libexec/src/runtime/netpoll.go:173 +0x57
	internal/poll.(*pollDesc).wait(0xc4202c2618, 0x72, 0xffffffffffffff00, 0x1f59660, 0x1f4f4f0)
		/usr/local/Cellar/go/1.9.2/libexec/src/internal/poll/fd_poll_runtime.go:85 +0xae
	internal/poll.(*pollDesc).waitRead(0xc4202c2618, 0xc42027a000, 0x1000, 0x1000)
		/usr/local/Cellar/go/1.9.2/libexec/src/internal/poll/fd_poll_runtime.go:90 +0x3d
	internal/poll.(*FD).Read(0xc4202c2600, 0xc42027a000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
		/usr/local/Cellar/go/1.9.2/libexec/src/internal/poll/fd_unix.go:126 +0x18a
	net.(*netFD).Read(0xc4202c2600, 0xc42027a000, 0x1000, 0x1000, 0x1, 0x2, 0xc42050b788)
		/usr/local/Cellar/go/1.9.2/libexec/src/net/fd_unix.go:202 +0x52
	net.(*conn).Read(0xc4201a0068, 0xc42027a000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
		/usr/local/Cellar/go/1.9.2/libexec/src/net/net.go:176 +0x6d
	crypto/tls.(*block).readFromUntil(0xc4205c8840, 0x250ebe8, 0xc4201a0068, 0x5, 0xc4201a0068, 0xc4202fc480)
		/usr/local/Cellar/go/1.9.2/libexec/src/crypto/tls/conn.go:488 +0x95
	crypto/tls.(*Conn).readRecord(0xc42004b500, 0x1a8d517, 0xc42004b620, 0x102d7fb)
		/usr/local/Cellar/go/1.9.2/libexec/src/crypto/tls/conn.go:590 +0xe0
	crypto/tls.(*Conn).Read(0xc42004b500, 0xc420292000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
		/usr/local/Cellar/go/1.9.2/libexec/src/crypto/tls/conn.go:1134 +0x110
	net/http.(*persistConn).Read(0xc4200d58c0, 0xc420292000, 0x1000, 0x1000, 0xc42028dfa0, 0xc420294058, 0x1053cb0)
		/usr/local/Cellar/go/1.9.2/libexec/src/net/http/transport.go:1391 +0x140
	bufio.(*Reader).fill(0xc42029cf60)
		/usr/local/Cellar/go/1.9.2/libexec/src/bufio/bufio.go:97 +0x11a
	bufio.(*Reader).Peek(0xc42029cf60, 0x1, 0x0, 0x0, 0x0, 0xc420294360, 0x0)
		/usr/local/Cellar/go/1.9.2/libexec/src/bufio/bufio.go:129 +0x3a
	net/http.(*persistConn).readLoop(0xc4200d58c0)
		/usr/local/Cellar/go/1.9.2/libexec/src/net/http/transport.go:1539 +0x185
	created by net/http.(*Transport).dialConn
		/usr/local/Cellar/go/1.9.2/libexec/src/net/http/transport.go:1186 +0xa2e

	goroutine 643 [select]:
	net/http.(*persistConn).writeLoop(0xc4200d58c0)
		/usr/local/Cellar/go/1.9.2/libexec/src/net/http/transport.go:1759 +0x165
	created by net/http.(*Transport).dialConn
		/usr/local/Cellar/go/1.9.2/libexec/src/net/http/transport.go:1187 +0xa53

	goroutine 666 [chan receive]:
	code.cloudfoundry.org/cli/util/configv3.catchSignal(0xc42029d080, 0xc420256c00, 0x27)
		/Users/worker/concourse/volumes/live/9f25a4ac-9961-42bd-7328-16fe28719986/volume/gopath/src/code.cloudfoundry.org/cli/util/configv3/config.go:261 +0x39
	created by code.cloudfoundry.org/cli/util/configv3.WriteConfig
		/Users/worker/concourse/volumes/live/9f25a4ac-9961-42bd-7328-16fe28719986/volume/gopath/src/code.cloudfoundry.org/cli/util/configv3/config.go:245 +0x2f1

	goroutine 724 [select]:
	code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb%2ev1.(*ProgressBar).refresher(0xc420122420)
		/Users/worker/concourse/volumes/live/9f25a4ac-9961-42bd-7328-16fe28719986/volume/gopath/src/code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb.v1/pb.go:462 +0x109
	created by code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb%2ev1.(*ProgressBar).Start
		/Users/worker/concourse/volumes/live/9f25a4ac-9961-42bd-7328-16fe28719986/volume/gopath/src/code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb.v1/pb.go:123 +0xc0

	goroutine 707 [select]:
	code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb%2ev1.(*ProgressBar).refresher(0xc4200a6160)
		/Users/worker/concourse/volumes/live/9f25a4ac-9961-42bd-7328-16fe28719986/volume/gopath/src/code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb.v1/pb.go:462 +0x109
	created by code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb%2ev1.(*ProgressBar).Start
		/Users/worker/concourse/volumes/live/9f25a4ac-9961-42bd-7328-16fe28719986/volume/gopath/src/code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb.v1/pb.go:123 +0xc0

	goroutine 728 [select, locked to thread]:
	runtime.gopark(0x1a8d4f0, 0x0, 0x1a491a8, 0x6, 0x18, 0x1)
		/usr/local/Cellar/go/1.9.2/libexec/src/runtime/proc.go:287 +0x12c
	runtime.selectgo(0xc420142750, 0xc4202944e0)
		/usr/local/Cellar/go/1.9.2/libexec/src/runtime/select.go:395 +0x1149
	runtime.ensureSigM.func1()
		/usr/local/Cellar/go/1.9.2/libexec/src/runtime/signal_unix.go:511 +0x1fe
	runtime.goexit()
		/usr/local/Cellar/go/1.9.2/libexec/src/runtime/asm_amd64.s:2337 +0x1

	goroutine 663 [select]:
	code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb%2ev1.(*ProgressBar).refresher(0xc420150000)
		/Users/worker/concourse/volumes/live/9f25a4ac-9961-42bd-7328-16fe28719986/volume/gopath/src/code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb.v1/pb.go:462 +0x109
	created by code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb%2ev1.(*ProgressBar).Start
		/Users/worker/concourse/volumes/live/9f25a4ac-9961-42bd-7328-16fe28719986/volume/gopath/src/code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb.v1/pb.go:123 +0xc0
 1.19 MiB / 1.19 MiB [=============================================================================================================================] 100.00%

		Your Platform Details
		e.g. Mac OS X 10.11, Windows 8.1 64-bit, Ubuntu 14.04.3 64-bit

		Shell
		e.g. Terminal, iTerm, Powershell, Cygwin, gnome-terminal, terminator
```

The environment in question might be a bit broken, but it shouldn't panic in this manner. Here's the debug log around the upload:
```
DEBU[0003] archive created                               archivePath=/var/folders/qx/dfk6jyx17y54jk5cw7x177gw0000gn/T/cf-cli-589684100
INFO[0003] uploading archive
DEBU[0003] uploading app bits                            appGUID=a463673b-4799-45d4-995f-61ee042e30e7 archiveSize=1247888
INFO[0003] received apply event: creating archive
Packaging files to upload...
INFO[0003] received apply event: uploading application with archive
Uploading files...
DEBU[0003] starting progress bar
 1.19 MiB / 1.19 MiB [=============================================================================================================================] 100.00%ERRO[0004] streaming archive: error seeking a stream on retry: Unexpected Response
Response code: 500
CC code:       0
CC error code:
Request ID:    790a1521-87f2-4214-60b7-3617ee263714::d1d30247-3108-4de1-bcb2-e5e75587111f
Description:   {
  "error_code": "UnknownError",
  "description": "An unknown error occurred.",
  "code": 10001
}
  archivePath=/var/folders/qx/dfk6jyx17y54jk5cw7x177gw0000gn/T/cf-cli-589684100
INFO[0004] received apply event: retry upload
INFO[0004] uploading archive
Retrying upload due to an error...
DEBU[0004] uploading app bits                            appGUID=a463673b-4799-45d4-995f-61ee042e30e7 archiveSize=1247888
INFO[0004] received apply event: uploading application with archive
Uploading files...
DEBU[0004] starting progress bar
 1.19 MiB / 1.19 MiB [=============================================================================================================================] 100.00%ERRO[0005] streaming archive: error seeking a stream on retry: Unexpected Response
Response code: 500
CC code:       0
CC error code:
Request ID:    2d792927-9dc6-4d2e-412c-bfe9339a2be0::efd89028-5562-4c51-9b07-596c33c393bc
Description:   {
  "error_code": "UnknownError",
  "description": "An unknown error occurred.",
  "code": 10001
}
  archivePath=/var/folders/qx/dfk6jyx17y54jk5cw7x177gw0000gn/T/cf-cli-589684100
INFO[0005] received apply event: retry upload
Retrying upload due to an error...
INFO[0005] uploading archive
DEBU[0005] uploading app bits                            appGUID=a463673b-4799-45d4-995f-61ee042e30e7 archiveSize=1247888
INFO[0005] received apply event: uploading application with archive
Uploading files...
DEBU[0005] starting progress bar
 1.19 MiB / 1.19 MiB [=============================================================================================================================] 100.00%ERRO[0006] streaming archive: error seeking a stream on retry: Unexpected Response
Response code: 500
CC code:       0
CC error code:
Request ID:    b7281f52-749f-4e6a-7be6-8d60ab9b93cc::779ffe69-3c4f-4fee-8c45-6f9c4677624d
Description:   {
  "error_code": "UnknownError",
  "description": "An unknown error occurred.",
  "code": 10001
}
  archivePath=/var/folders/qx/dfk6jyx17y54jk5cw7x177gw0000gn/T/cf-cli-589684100
INFO[0006] received apply event: retry upload
Retrying upload due to an error...
ERRO[0006] process apply stream: upload failed
```

-- Also filed with @XenoPhex

**Comment by XenoPhex** [comment]
Fixed on master, will close on release.

**Comment by XenoPhex** [comment]
Fixed in 6.35.*
